packages/client/src/components/app/blocks/TableBlock.svelte

```diff
@@ -109,6 +113,10 @@
     const binding = `${prefix}.${safe(primaryDisplay)}`
     return `{{#if ${binding}}}{{${binding}}}{{else}}Details{{/if}}`
   }
+
+  const getSortColumn = () => {
+    return sortColumn || (isDisplayFormula ? undefined : primaryDisplay)
```

Comment:
Thanks for taking a look - good shout!

I don't think your logic was quite right. If the `sortColumn` is set by the user, then you always want to use that column no matter what the primary display is. 
Would you mind having another glance over to be sure you're happy?

Reply:
Ah in that case then we need to maintain the original value, which has been accounted for with:
```js
 if (sortColumn) {
      return sortColumn
}
```

This looks better now 👍 although it seems according to @aptkingston this could have other ramifications if it's done in the FE.